internal/kafka/internal/handlers/admin_kafka.go

```diff
@@ -138,18 +141,89 @@ func (h adminKafkaHandler) Update(w http.ResponseWriter, r *http.Request) {
 				if kafkaRequest.DesiredKafkaVersion != kafkaUpdateReq.KafkaVersion && kafkaUpdateReq.KafkaVersion != "" && kafkaRequest.KafkaUpgrading {
 					return errors.New(errors.ErrorValidation, "Unable to update kafka version. Another upgrade is already in progress.")
 				}
+
+				currentKafkaVersion, _ := arrays.FirstNonEmpty(kafkaRequest.ActualKafkaVersion, kafkaRequest.DesiredKafkaVersion)
+
+				vCompKafka, ek := api.CompareSemanticVersionsMajorAndMinor(currentKafkaVersion, kafkaRequest.DesiredKafkaVersion)
+
+				if ek != nil {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to compare desired kafka version: %s with actual kafka version: %s", kafkaRequest.DesiredKafkaVersion, currentKafkaVersion))
+				}
+
+				// no minor/ major version downgrades allowed for kafka version
+				if vCompKafka > 0 {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to downgrade kafka: %s version: %s to the following kafka version: %s", kafkaRequest.ID, currentKafkaVersion, kafkaRequest.DesiredKafkaVersion))
+				}
+
 				return nil
 			},
 			func() *errors.ServiceError { // Validate DesiredStrimziVersion
 				if kafkaRequest.DesiredStrimziVersion != kafkaUpdateReq.StrimziVersion && kafkaUpdateReq.StrimziVersion != "" && kafkaRequest.StrimziUpgrading {
 					return errors.New(errors.ErrorValidation, "Unable to update strimzi version. Another upgrade is already in progress.")
 				}
+
 				return nil
 			},
 			func() *errors.ServiceError { // Validate DesiredKafkaIBPVersion
 				if kafkaRequest.DesiredKafkaIBPVersion != kafkaUpdateReq.KafkaIbpVersion && kafkaUpdateReq.KafkaIbpVersion != "" && kafkaRequest.KafkaIBPUpgrading {
 					return errors.New(errors.ErrorValidation, "Unable to update ibp version. Another upgrade is already in progress.")
 				}
+
+				currentIBPVersion, _ := arrays.FirstNonEmpty(kafkaRequest.ActualKafkaIBPVersion, kafkaRequest.DesiredKafkaIBPVersion)
```

Comment:
This is using the old `DesiredVersion` from the database but it should be using the DesiredVersion from the request.

This is cause of the issue.
The desired versions are set in the https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/1799599ca3cd34f8383a59d11111352b9951a9f1/internal/kafka/internal/handlers/admin_kafka.go#L236-L247 which happens after the validations have been run

Reply:
To fix the integration issue we need to use the `kafkaUpdateReq.xxx` if it is set.

e.g `kafkaUpdateReq.StrimziVersion` instead of `kafkaRequest.DesiredKafkaVersion` for this line specifically